src/commands/cmd_hash.cc

```diff
@@ -44,10 +44,22 @@ class CommandHGet : public Commander {
 
 class CommandHSetNX : public Commander {
  public:
+  Status Parse(const std::vector<std::string> &args) override {
+    if (args.size() % 2 != 0) {
+      return {Status::RedisParseErr, errWrongNumOfArguments};
+    }
+    return Commander::Parse(args);
+  }
+
   Status Execute(Server *svr, Connection *conn, std::string *output) override {
+    std::vector<FieldValue> field_values;
+    for (size_t i = 2; i < args_.size(); i += 2) {
+      field_values.emplace_back(FieldValue{args_[i], args_[i + 1]});
```

Comment:
```suggestion
      field_values.emplace_back(args_[i], args_[i + 1]);
```

Reply:
BTW I think it is more proper to put the for loop to `Parse`.

Reply:
@PragmaTwice Thank you for your reasonable suggestions.

Reply:
Done